Remove non-prod schedule variables

adampie opened the pull request:
First MR switching multi-env schedule logic and variables into a single environment/variable.

adampie committed:
Simplify scheduling logic by removing non-prod schedules

Removed conditional logic for non-prod scheduling and related variables to streamline environment management. All schedules now use uniform variables regardless of environment, reducing complexity and potential errors.

paulschwarzenberger reviewed (commented):
I agree that we need to take the references to `local.env` out of the module, however we also need to record these variations between prod and nonprod environments so that appropriate configurations and / or conditionals can be included in the code which calls the module.

I'd suggest starting an examples folder - which is in any case required to be compliant with the Terraform registry - and include two examples, one the default which I'd suggest should be nonprod and then another example passing in the variables for prod.

The default values should therefore be the nonprod ones, as that's more straightforward for people to start with.

adampie committed:
Update scan and update schedules in variables.tf

adampie committed:
Remove examples tfvars file

adampie committed:
Examples init

adampie committed:
Add providers and schedules to nonprod, prod, and org examples

Added HashiCorp providers (aws, archive, null, random) configurations to nonprod, prod, and organizations terraform files. Updated module configurations for domain_protect with scan and update schedules. Set IAM role for domain_protect in organizations example and changed default value of takeover variable to false.

nitrocode commented on examples/nonprod/main.tf:
Let's rename dir to `examples/complete/main.tf`

nitrocode commented on examples/organizations/terraform.tf:
```diff
@@ -0,0 +1,60 @@
+      version = "~> 3.1.0"
+    }
+  }
+}
```
Can we add this as `versions.tf` ?

nitrocode commented on examples/organizations/terraform.tf:
```diff
@@ -0,0 +1,60 @@
+
+provider "archive" {}
+provider "null" {}
+provider "random" {}
```
Can we add this as `providers.tf` ?

nitrocode commented on examples/prod/README.md:
```diff
@@ -0,0 +1 @@
+# Example deployment to a prod environment
```
Instead of `prod` and `nonprod`, we can use `complete` as the dir name and then use `tfvars` file for each environment or even use a single tfvars file with a map of each environment and then a `for_each` on the module per environment

nitrocode committed:
chore: merge and fix conflicts

Signed-off-by: nitrocode <7775707+nitrocode@users.noreply.github.com>

nitrocode commented on variables.tf:
```diff
@@ -90,7 +72,7 @@ variable "lambdas" {
 variable "takeover" {
   description = "Create supported resource types to prevent malicious subdomain takeover"
-  default     = true
+  default     = false
```
How come this was toggled to `false` ?

paulschwarzenberger replied:
This is supposed to be `false` by default, i.e. in the dev environment, and `true` by default for production.

The reason for this is to prevent both a dev and a prod environment trying to take over the same S3 bucket / ElasticBeanstalk resulting in conflicts.

nitrocode replied:
Understood, thank you

nitrocode committed:
chore: consolidate examples

Signed-off-by: nitrocode <7775707+nitrocode@users.noreply.github.com>

nitrocode committed:
chore: newline

Signed-off-by: nitrocode <7775707+nitrocode@users.noreply.github.com>

nitrocode commented on variables.tf:
```diff
@@ -41,25 +41,13 @@ variable "reports_schedule" {
-variable "scan_schedule_nonprod" {
-  description = "schedule for running domain-protect scans in non-prod, reduced to save costs, e.g. 12 hours"
+  description = "schedule for running domain-protect scans, e.g. 24 hours"
   default     = "24 hours"
```
Instead of defaulting it to all the `nonprod` values. Shouldn't we default it to all the `prod` values since those are already the existing defaults ?

adampie replied:
https://owasp.slack.com/archives/C04BPJ5B2P4/p1724159709558629?thread_ts=1724097809.543719&channel=C04BPJ5B2P4&message_ts=1724159709.558629

nitrocode replied:
Make sense. Thanks

nitrocode reviewed (commented):
Added some comments (Unsure why I have to submit a review to add comments now)

nitrocode committed:
chore: rm slack and takeover dev env

Signed-off-by: nitrocode <7775707+nitrocode@users.noreply.github.com>
